serde/src/java/org/apache/hadoop/hive/serde2/lazybinary/LazyBinaryStruct.java

```diff
@@ -221,7 +221,7 @@ public void init(BinaryComparable src) {
       fieldBytes = src.getBytes();
       int length = src.getLength();
       byte nullByte = fieldBytes[0];
-      int lastFieldByteEnd = 1, fieldStart = -1, fieldLength = -1;
```

Comment:
fieldStart and fieldLength are not intented to be used as local variable; they should be set to proper values in init() so that the caller of getShort() retrieves correct field from the BinaryStruct.
I split the field declaration into separate lines.